### DISCUSSION TITLE
Mantine v7 + SSR guide?

### DISCUSSION THREAD

**Discussion by d9k** [mantinedev/mantine#5493] [Q&A]
[Discord theme](https://discord.com/channels/854810300876062770/1153906078271156304/1153906078271156304) says Mantine v7 supports SSR.

But where can be found SSR documentation page like it was for `v5` version ( :fallen_leaf: [Server side rendering | Mantine v5](https://v5.mantine.dev/guides/ssr/) )?

I tried naive approach with implementing SSR (just duplicate `<MantineProvider>` at both server and client) and got 

```
Prop `id` did not match. Server: "mantine-R1brfde" Client: "mantine-Raurrb" at input
```

warning on page with form.

Mantine version: `7.1.2`.

BTW when I force reload tab or event after restart project server ids in the error message are the same.

Form field code:

```tsx
<TextInput
  required
  size={inputSize}
  placeholder='email'
  radius='md'
  {...form.getInputProps('email')}
/>
```

**Comment by rtivital** [comment]
Mantine supports server side rendering, you can find guides for popular frameworks on the getting started page

**Reply by d9k** [reply]
@rtivital, thanks for the answer! 

I use Ultra framework for [Deno](https://deno.com): https://ultrajs.dev, [(GH)](https://github.com/exhibitionist-digital/ultra)

Just added `id` to `<TextInput>` params and warning gone:
https://github.com/d9k/citations-supabase-demo/blob/main/src/features/auth/OtpForm.tsx

But if I don't provide constant id to `<TextInput>` it seems it changes every form rerender.

**Reply by d9k** [reply]
@rtivital, it would be nice to have guide on implementing Mantine SSR on [express](https://expressjs.com) Node.JS framework, because it the most well-known framework among Node.JS & Deno developers and thus SSR implementation there can be easy understood and ported to any new framework.

**Reply by rtivital** [reply]
There is not any logic in Mantine that is specific to SSR. All components render to string out of the box.

**Reply by kishanio** [reply]
But what about the styles that ends up in the head. I believe this is causing flicker when rendered on server.

**Reply by rtivital** [reply]
In v7 styles are distributed via a css file, no styles are inserted to head.
